packages/netsuite-adapter/test/change_validators/extra_reference_dependencies.test.ts

```diff
@@ -0,0 +1,133 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { ElemID, InstanceElement, ObjectType, ReferenceExpression, toChange } from '@salto-io/adapter-api'
+import { fileType } from '../../src/types/file_cabinet_types'
+import { customsegmentType } from '../../src/autogen/types/standard_types/customsegment'
+import extraReferenceDependenciesValidator from '../../src/change_validators/extra_reference_dependencies'
+import { CUSTOM_RECORD_TYPE, METADATA_TYPE, NETSUITE, PATH, SCRIPT_ID } from '../../src/constants'
+
+
+describe('extra reference changes', () => {
+  const customsegment = customsegmentType().type
+
+  const fileInstance = new InstanceElement('file_instance', fileType(), {
+    [PATH]: 'Templates/E-mail Templates/Inner EmailTemplates Folder/content.html',
+  })
+
+  const customRecordType = new ObjectType({
+    elemID: new ElemID(NETSUITE, 'custom_record_type'),
+    annotations: {
+      [METADATA_TYPE]: CUSTOM_RECORD_TYPE,
+      [SCRIPT_ID]: 'custom_record_type_script_id',
+    },
+  })
+
+  const customSegmentInstance = new InstanceElement('custom_segment_instance', customsegment, {
+    [SCRIPT_ID]: 'custom_segment_instance_script_id',
+    description: new ReferenceExpression(
+      fileInstance.elemID.createNestedID(PATH),
+      fileInstance.value[PATH],
+      fileInstance
+    ),
+  })
+
+  customSegmentInstance.value.recordtype = new ReferenceExpression(
+    customRecordType.elemID.createNestedID('attr', SCRIPT_ID), 'val', customRecordType
+  )
+
+  customRecordType.annotations.customsegment = new ReferenceExpression(
+    customSegmentInstance.elemID.createNestedID(SCRIPT_ID),
+    customSegmentInstance.value[SCRIPT_ID],
+    customSegmentInstance
+  )
+
+  const dependsOn2Instances = new InstanceElement('depends_on_2_instances', customsegment, {
+    [SCRIPT_ID]: 'depends_on_2_instances_script_id',
+    label: new ReferenceExpression(
+      customSegmentInstance.elemID.createNestedID(SCRIPT_ID),
+      customSegmentInstance.value[SCRIPT_ID],
+      customSegmentInstance
+    ),
+    description: new ReferenceExpression(
+      fileInstance.elemID.createNestedID(PATH),
+      fileInstance.value[PATH],
+      fileInstance
+    ),
+  })
+
+  // dependsOn2Instances ---> customSegmentInstance <---> customRecordType
+  //        |                     |
+  //        '-->  fileInstance <--'
+
+  it('should not have ChangeError when deploying an instance with its dependencies, when requesting to add only required references', async () => {
+    const changeErrors = await extraReferenceDependenciesValidator([
+      toChange({ before: customRecordType, after: customRecordType }),
+      toChange({ after: customSegmentInstance }),
+      toChange({ before: fileInstance, after: fileInstance }),
```

Comment:
maybe change it to `dependsOn2Instances` to see that no warnings are added:
```suggestion
      toChange({ before: dependsOn2Instances, after: dependsOn2Instances }),
```